lib/AST/ASTContext.cpp

```diff
@@ -8191,6 +8191,225 @@ QualType ASTContext::mergeObjCGCQualifiers(QualType LHS, QualType RHS) {
   return QualType();
 }
 
+//===--------------------------------------------------------------------===//
+//                    Predicates For Checked C checked types
+//===--------------------------------------------------------------------===//
+
+static bool lessThan(bool Self, bool Other) {
+  return (Self != Other && !Self);
+}
+
+bool ASTContext::isAtLeastAsCheckedAs(QualType T1, QualType T2) const {
+  if (T1.isNull() || T2.isNull())
+    return true;
```

Comment:
You can have an "empty" QualType that has a null Type * associated with it.   We do need to bail out if we see that (the varaibles at lines 8212 or 8213 might end up with null pointers).    I think I should return false in this case (a null type is never at least as checked as anything, because we know nothing about it).